pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -645,3 +625,17 @@ func (ts *TaskSpec) GetIndexingReferencesToArrayParams() sets.String {
 	}
 	return sets.NewString(arrayIndexParamRefs...)
 }
+
+// ValidateBetaFeaturesEnabledForParamArrayIndexing validates that "enable-api-fields" is set to "alpha" or "beta" if the task spec
+// contains indexing references to array params.
+// This can be removed when array param indexing is moved to "stable".
+func (ts *TaskSpec) ValidateBetaFeaturesEnabledForParamArrayIndexing(ctx context.Context) *apis.FieldError {
+	if config.CheckAlphaOrBetaAPIFields(ctx) {
+		return nil
+	}
```

Comment:
this is a bit different from the function it's replacing?

Reply:
Yes-- see the other comment I made; it's the same reason.